python/tests/test_stats.py

```diff
@@ -384,7 +384,123 @@ def naive_genealogical_nearest_neighbours(ts, focal, reference_sets):
     return A
 
 
-class TestGenealogicalNearestNeighbours:
+def parse_time_windows(ts, time_windows):
+    if time_windows is None:
+        time_windows = [0.0, ts.max_root_time]
+    return np.array(time_windows)
+
+
+def windowed_genealogical_nearest_neighbours(
+    ts,
+    focal,
+    reference_sets,
+    windows=None,
+    time_windows=None,
+    span_normalise=True,
+    time_normalise=True,
+):
+    """
+    genealogical_nearest_neighbours with support for span- and time-based windows
+    """
+    reference_set_map = np.zeros(ts.num_nodes, dtype=int) - 1
```

Comment:
Helps readability to be specific about what the `-1`s are:
```suggestion
    reference_set_map = np.full(ts.num_nodes, tskit.NULL, dtype=int)
```